### PR TITLE
Filter name

### PR DESCRIPTION
**Pull request by DiegoTheExplorar**

### CODE REVIEW

File: src/test/java/seedu/address/logic/commands/FilterCommandTest.java [hunk 1]
```diff
@@ -4,6 +4,7 @@
 import static org.junit.jupiter.api.Assertions.assertFalse;
 import static org.junit.jupiter.api.Assertions.assertTrue;
 import static seedu.address.testutil.TypicalPersons.BENSON;
+import static seedu.address.testutil.TypicalPersons.CARL;
```
Comment (valhrd) [inline src/test/java/seedu/address/logic/commands/FilterCommandTest.java:7]: A new name, how cool!

File: src/test/java/seedu/address/logic/parser/FilterCommandParserTest.java [hunk 2]
```diff
@@ -46,13 +47,35 @@ public void parse_emptyTag_throwsParseException() {
...
+    public void parse_emptyInput_throwsParseException() {
+        assertParseFailure(parser, "     ",
+                String.format(MESSAGE_INVALID_COMMAND_FORMAT, FilterCommand.MESSAGE_USAGE));
+    }
+
+    @Test
+    public void parse_invalidPrefix_throwsParseException() {
+        assertParseFailure(parser, " x/Alice",
+                String.format(MESSAGE_INVALID_COMMAND_FORMAT, FilterCommand.MESSAGE_USAGE));
+    }
```
Comment (valhrd) [inline src/test/java/seedu/address/logic/parser/FilterCommandParserTest.java:80]: Excellent test coverage!

File: src/test/java/seedu/address/logic/commands/FilterCommandTest.java [hunk 3]
```diff
@@ -53,70 +56,74 @@ public void equals() {
...
     @Test
     public void toStringMethod() {
-        Set<Tag> tags = new HashSet<>();
-        tags.add(new Tag("client"));
-        FilterCommand filterCommand = new FilterCommand("John", tags);
-        String expected = FilterCommand.class.getCanonicalName() + "{name=John, tags=[[client]]}";
+        Set<String> names = new HashSet<>(Arrays.asList("John", "Doe"));
+        Set<Tag> tags = new HashSet<>(Arrays.asList(new Tag("client")));
+        FilterCommand filterCommand = new FilterCommand(names, tags);
+        String expected = FilterCommand.class.getCanonicalName() + "{names=[John, Doe], tags=[[client]]}";
```
Comment (valhrd) [inline src/test/java/seedu/address/logic/commands/FilterCommandTest.java:126]: Very well written code!

File: src/test/java/seedu/address/logic/commands/FilterCommandTest.java [hunk 4]
```diff
@@ -53,70 +56,74 @@ public void equals() {
...
+    @Test
+    public void execute_multipleNames_personsFound() {
+        Set<String> names = new HashSet<>(Arrays.asList("Meier", "Kurz"));
+        String expectedMessage = FilterCommand.constructSuccessMessage(names, new HashSet<>());
+        FilterCommand command = new FilterCommand(names, new HashSet<>());
+        CommandResult result = command.execute(model);
+        assertEquals(expectedMessage, result.getFeedbackToUser());
+        assertEquals(Arrays.asList(BENSON, CARL, DANIEL), model.getFilteredPersonList());
+    }
+
```
Comment (valhrd) [inline src/test/java/seedu/address/logic/commands/FilterCommandTest.java:120]: Good test coverage!

File: src/main/java/seedu/address/logic/commands/FilterCommand.java [hunk 5]
```diff
@@ -26,32 +26,30 @@ public class FilterCommand extends Command {
...
 
-    private final String name;
+    private final Set<String> names;
     private final Set<Tag> tags;
 
     /**
      * Constructs a FilterCommand.
      *
-     * @param name The name to filter by.
+     * @param names The set of name to filter by.
```
Comment (valhrd) [inline src/main/java/seedu/address/logic/commands/FilterCommand.java:35]: Good specification!

File: src/main/java/seedu/address/logic/commands/FilterCommand.java [hunk 6]
```diff
@@ -26,32 +26,30 @@ public class FilterCommand extends Command {
...
     public CommandResult execute(Model model) {
         requireNonNull(model);
 
         Predicate<Person> predicate = person -> {
-            // Check if the person's name matches the filter name (if provided)
-            boolean nameMatches = name == null || name.isEmpty()
-                    || person.getName().toString().toLowerCase().contains(name.toLowerCase());
+            // Check if the person's name matches any of the filter names (if provided)
+            boolean nameMatches = names.isEmpty() || names.stream()
+                    .anyMatch(name -> person.getName().toString().toLowerCase().contains(name.toLowerCase()));
```
Comment (valhrd) [inline src/main/java/seedu/address/logic/commands/FilterCommand.java:49]: I really like your usage of Java Streams!

File: src/main/java/seedu/address/logic/commands/FilterCommand.java [hunk 7]
```diff
@@ -65,23 +63,27 @@ public CommandResult execute(Model model) {
...
         StringBuilder messageBuilder = new StringBuilder("Displaying filtered results: ");
-        if (name != null && !name.isEmpty()) {
-            messageBuilder.append("Name: ").append(name).append(" ");
+
+        if (!names.isEmpty()) {
+            String namesString = String.join(", ", names);
+            messageBuilder.append("Names: ").append(namesString).append(" ");
         }
+
         if (!tags.isEmpty()) {
```
Comment (valhrd) [inline src/main/java/seedu/address/logic/commands/FilterCommand.java:80]: Super clean, LGTM!

File: src/main/java/seedu/address/logic/commands/FilterCommand.java [hunk 8]
```diff
@@ -26,32 +26,30 @@ public class FilterCommand extends Command {
...
     @Override
     public CommandResult execute(Model model) {
         requireNonNull(model);
 
         Predicate<Person> predicate = person -> {
-            // Check if the person's name matches the filter name (if provided)
-            boolean nameMatches = name == null || name.isEmpty()
-                    || person.getName().toString().toLowerCase().contains(name.toLowerCase());
+            // Check if the person's name matches any of the filter names (if provided)
+            boolean nameMatches = names.isEmpty() || names.stream()
```
Comment (appleshill) [inline src/main/java/seedu/address/logic/commands/FilterCommand.java:48]: Neat solution! Great use of Java streams

File: src/main/java/seedu/address/logic/commands/FilterCommand.java [hunk 9]
```diff
@@ -65,23 +63,27 @@ public CommandResult execute(Model model) {
...
      * Constructs the success message based on the name and tags.
      */
-    public static String constructSuccessMessage(String name, Set<Tag> tags) {
+    public static String constructSuccessMessage(Set<String> names, Set<Tag> tags) {
         StringBuilder messageBuilder = new StringBuilder("Displaying filtered results: ");
-        if (name != null && !name.isEmpty()) {
-            messageBuilder.append("Name: ").append(name).append(" ");
+
+        if (!names.isEmpty()) {
+            String namesString = String.join(", ", names);
```
Comment (appleshill) [inline src/main/java/seedu/address/logic/commands/FilterCommand.java:76]: Great idea

File: src/test/java/seedu/address/logic/commands/FilterCommandTest.java [hunk 10]
```diff
@@ -53,70 +56,74 @@ public void equals() {
...
 
-    // Test case to test partial string filtering
     @Test
     public void execute_partialNameMatch_personsFound() {
-        Set<Tag> expectedTags = new HashSet<>();
-        String expectedMessage = FilterCommand.constructSuccessMessage("er", expectedTags);
-        FilterCommand command = new FilterCommand("er", expectedTags);
+        Set<String> names = new HashSet<>(Arrays.asList("er"));
+        String expectedMessage = FilterCommand.constructSuccessMessage(names, new HashSet<>());
+        FilterCommand command = new FilterCommand(names, new HashSet<>());
```
Comment (appleshill) [inline src/test/java/seedu/address/logic/commands/FilterCommandTest.java:105]: Good use of Hashsets.

**Review by appleshill — APPROVED** [review]

**Pull request merged**